cmd/soroban-rpc/internal/test/cli_test.go

```diff
@@ -211,13 +213,40 @@ func TestCLIRestore(t *testing.T) {
 		),
 	)
 
-	newExpirationSeq := getExpirationForLedgerEntry(t, client, getExpirationKey(t, getCounterLedgerKey(parseContractStrKey(t, strkeyContractID))))
+	newExpirationSeq := getExpirationForLedgerEntry(t, client, getCounterLedgerKey(parseContractStrKey(t, strkeyContractID)))
 	assert.Greater(t, newExpirationSeq, initialExpirationSeq)
 	assert.Equal(t, fmt.Sprintf("New expiration ledger: %d", newExpirationSeq), restoreOutput)
+
+	// test to see that we get an error when requesting the expiration ledger entry explicitly.
+	ledgerExpirationEntry := getExpirationKey(t, getCounterLedgerKey(parseContractStrKey(t, strkeyContractID)))
+	ledgerExpirationEntryB64, err := xdr.MarshalBase64(ledgerExpirationEntry)
+	require.NoError(t, err)
+	var getLedgerEntryResult methods.GetLedgerEntryResponse
+	err = client.CallResult(context.Background(), "getLedgerEntry", methods.GetLedgerEntryRequest{
+		Key: ledgerExpirationEntryB64,
+	}, &getLedgerEntryResult)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), methods.ErrLedgerExpirationEntriesCannotBeQueriedDirectly)
+
+	// repeat with getLedgerEntries
```

Comment:
is this verification of rpc `getLedgerEntry/Entries` to check `ErrLedgerExpirationEntriesCannotBeQueriedDirectly`  related to the cli restore test, it's fine here, just seems like it is a separate test appended to the end, noticed there is also a get_ledger_entries_test.go, maybe this could be a test in there.

I was looking to see if the nil case of LedgerEntryResult.ExpirationLedger is validated, i.e. if you request an Account's LedgerEntry, the go response should have LedgerEntryResult.ExpirationLedger=nil and the json response should not have `expirationLedgerSeq ` present, correct?

Reply:
That's correct for today. It might change in the future, and completely depends on the txmeta.

I'm not sure how to (easily) cover programmatically all the ledger entry types.